src/index.js

```diff
@@ -1,54 +1,145 @@
+/**
+ * Query github for semantic version tag.
+ *
+ * @typedef {{tag: string, version: string, hash: string, date: Date}} Tag
+ */
+
 'use strict';
 
 const semver = require('semver');
 const childProcess = require('child-process-promise');
 
 const tagRegex = /tag:\s*([^,)]+)/g;
-const commitDetailsRegex = /.*?(tag:\s*([^,)]+).*);(.+);(.+)/;
+const commitDetailsRegex = /^(.+);(.+);(.+)$/;
 
+/**
+ * Run shell command and resolve with stdout content
+ *
+ * @param  {string} command Shell command
+ * @return {Promise<string,Error>}
+ */
 function runCommand(command) {
   return childProcess.exec(command)
     .then(result => result.stdout);
 }
 
-function getFirstValidTag(allTags, guess) {
-  let match;
-  let result = guess;
-  while (match = tagRegex.exec(allTags)) {
-    if (semver.valid(match[1])) {
-      result = match[1];
-      break;
+/**
+ * Get all tags with a semantic version name out of a list of Refs
+ *
+ * @param  {string} refs List of refs
+ * @return {Arrays<Tag>}
+ */
+function getSemanticTags(refs) {
+  const tags = [];
+
+  while (true) {
+    const match = tagRegex.exec(refs);
+
+    if (match == null) {
+      return tags;
```

Comment:
Would you please refactor the function to return the result (tags) at the end, for readability purpose ?
It is not a long function, so my opinion may not be relevant, but if someone else updates the function it may become complicated very fast.

For example, something like 
```javascript
function getSemanticTags(refs) {
  const tags = [];

  while (true) {
    const match = tagRegex.exec(refs);

    if (match == null) {
      break;
    }

    const tag = match[1];
    const version = semver.valid(tag);

    if (version != null) {
      tags.push({ tag, version });
    }
  }

  return tags;
}
```

or (without the linting warning, but with line duplication):
```javascript
function getSemanticTags(refs) {
  const tags = [];
  let match = tagRegex.exec(refs);

  while (match) {
    const tag = match[1];
    const version = semver.valid(tag);

    if (version != null) {
      tags.push({ tag, version });
    }

    match = tagRegex.exec(refs);
  }

  return tags;
}
```